class.jetpack-gutenberg.php

```diff
@@ -155,109 +164,31 @@ static function load( $response = null, $handler = null, $request = null ) {
-
-		/**
-		 * Filter the list of block editor plugins that are available through jetpack.
-		 *
-		 * This filter is populated by Jetpack_Gutenberg::jetpack_set_available_blocks
-		 *
-		 * @since 6.9.0
-		 *
-		 * @param array
-		 */
-		self::$plugins = apply_filters( 'jetpack_set_available_plugins', self::$default_plugins );
-		self::set_blocks_availability();
-		self::set_plugins_availability();
-		self::register_blocks();
-		return $response;
-	}
-
-	static function is_registered( $slug ) {
-		return isset( self::$registered[ $slug ] );
+		self::$extensions = apply_filters( 'jetpack_set_available_blocks', self::get_jetpack_gutenberg_extensions_whitelist() );
 	}
 
-	static function get_registered_args( $slug ) {
-		return self::$registered[ $slug ]['args'];
-	}
-	static function get_registered_type( $slug ) {
-		return self::$registered[ $slug ]['type'];
-	}
-
-	static function is_available( $slug ) {
-		if ( ! isset( self::$availability[ $slug ]['available'] ) ) {
-			return false;
-		}
-		return (bool) self::$availability[ $slug ]['available'];
-	}
-
-
-	static function get_extension_availability( $slug ) {
-		if ( ! self::is_registered( $slug ) ) {
-			return array( 'available' => false, 'unavailable_reason' => 'missing_module' );
-		}
-
-		$availability = self::$registered[ $slug ]['availability'];
-
-		if ( isset( $availability['callback'] ) ) {
-			$availability = call_user_func( $availability['callback'] );
-		}
-
-		$availability['available'] = isset( $availability['available'] )
-			? (bool) $availability['available']
-			: true ; // this is the default value
-
-		if ( ! $availability['available'] ) {
-			$availability['unavailable_reason'] = isset( $availability['unavailable_reason'] )
-				? $availability['unavailable_reason']
-				: 'unknown'; //
-		}
-
-		return $availability;
-	}
-
-	static function set_blocks_availability() {
-		foreach ( self::$blocks as $slug ) {
-			self::$availability[ $slug ] = self::get_extension_availability( $slug );
-		}
-	}
-
-	static function set_plugins_availability() {
-		foreach ( self::$plugins as $slug ) {
-			self::$availability[ $slug ] = self::get_extension_availability( $slug );
-		}
-	}
-
-	static function register_blocks() {
-		/**
-		 * Filter the list of block editor plugins should be passed to register_block_type.
-		 * You can use this to manually register some blocks without Jetpack registering them for you.
-		 *
-		 * This filter is populated by Jetpack_Gutenberg::jetpack_set_available_blocks
-		 *
-		 * @since 6.9.0
-		 *
-		 * @param array
-		 */
-		$blocks = apply_filters( 'jetpack_blocks_to_register', self::$blocks );
-		foreach ( $blocks as $slug ) {
-			if ( self::is_available( $slug ) ) {
-				register_block_type( 'jetpack/' . $slug, self::get_registered_args( $slug ) );
-			}
-		}
+	/**
+	 * Resets the class to its original state
+	 *
+	 * Used in unit tests
+	 *
+	 * @return void
+	 */
+	public static function reset() {
```

Comment:
This is what happens when we use classes with only static methods.

Reply:
Indeed. I hated myself a tiny bit when writing this.

Reply:
See https://github.com/Automattic/jetpack/pull/11091#discussion_r246811623 which includes some thoughts on turning this class into an actually instantiable one.